From 86e55980e3025678882cb9c4c78614f86cdc1f04 Mon Sep 17 00:00:00 2001
From: Rodney Dawes <dobey.pwns@gmail.com>
Date: Tue, 9 Aug 2022 13:57:39 -0400
Subject: [PATCH] ut_mimserveroptions: Move the operator== definition to global
 namespace

With gcc-12 having the operator== in anonymous namespace for a struct which
is in the global namespace results in a lengthy compilation error. Moving
this non-mmember function into the global namespace fixes the issue.

Fixes #112
---
 tests/ut_mimserveroptions/ut_mimserveroptions.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/tests/ut_mimserveroptions/ut_mimserveroptions.cpp b/tests/ut_mimserveroptions/ut_mimserveroptions.cpp
index 7c904db..f533927 100644
--- a/tests/ut_mimserveroptions/ut_mimserveroptions.cpp
+++ b/tests/ut_mimserveroptions/ut_mimserveroptions.cpp
@@ -42,12 +42,12 @@ namespace {
                           "-testability", "TESTABILITY", "-qdevel", "-reverse",
                           "-stylesheet", "-widgetcount", "-qdebug",
                           "-software" } };
+}
 
-    bool operator==(const MImServerCommonOptions &x,
-                    const MImServerCommonOptions &y)
-    {
-        return (x.showHelp == y.showHelp);
-    }
+bool operator==(const MImServerCommonOptions &x,
+                const MImServerCommonOptions &y)
+{
+    return (x.showHelp == y.showHelp);
 }
 
 
-- 
2.37.1

